test/lit/binary/debug-bad-binary.test

```diff
@@ -38,4 +38,10 @@ RUN: not wasm-opt --debug %s.wasm 2>&1 | filecheck %s
 ;; CHECK-NEXT:    (i32.const 10)
 ;; CHECK-NEXT:   )
 ;; CHECK-NEXT:  )
+;; CHECK-NEXT:  (func $1
+;; CHECK-NEXT:   (unreachable)
+;; CHECK-NEXT:  )
+;; CHECK-NEXT:  (func $2
+;; CHECK-NEXT:   (unreachable)
+;; CHECK-NEXT:  )
```

Comment:
This seems like it could be confusing for users and even for us when debugging, to see these unreachables in the function bodies. How about storing nullptr for the body, and making Print handle those (by printing "placeholder" or such)?

Reply:
I was originally going that route, but Print isn't the only place that would need to change. ModuleUtils would need to update the type collection code not to trip over the nullptr, too.

Reply:
Collecting the types is a prerequisite for printing the module. Even when the module is not complete, it's still in a consistent enough state that we can walk it and collect types.

Reply:
I see. If it's only in those two places, perhaps it makes sense to do? I'd lean that way myself.